tests/cnf/core/network/sriov/tests/metricsExporter.go

```diff
@@ -0,0 +1,310 @@
+					workerNodeList[cWorkerindex].Object.Name, 0)
+				var serverResources = defineTestResources("server",
+					sriovInterfacesUnderTest[sPfindex], sriovDeviceID, "netdevice",
+					workerNodeList[sWorkerindex].Object.Name, 1)
+
+				cPod, _ := createTestResources(clientResources, serverResources)
+
+				By("ICMP check between client and server pods")
+				Eventually(func() error {
+					return cmd.ICMPConnectivityCheck(cPod, []string{tsparams.ServerIPv4IPAddress}, "net1")
+				}, 1*time.Minute, 2*time.Second).Should(Not(HaveOccurred()), "ICMP Failed")
+
+				By("Wait 1min for prometheus scraping")
+				time.Sleep(1 * time.Minute)
+
+				By("Verify RX and TX packets counters are >0")
+				Expect(fetchScalarFromPromQLoutput(execPromQLandReturnString(serverPodRXPromQL))).Should(BeNumerically(">", 0))
+				Expect(fetchScalarFromPromQLoutput(execPromQLandReturnString(serverPodTXPromQL))).Should(BeNumerically(">", 0))
+			},
+			Entry(" Same PF - Same Worker", 0, 0, 0, 0),
+			Entry(" Different PF - Same Worker", 0, 1, 0, 0),
+			Entry(" Different Worker", 0, 0, 0, 1),
+		)
+
+	})
+
+func defineTestResources(role, pfName, nicVendor, deviceType, workerNode string, vfRange int) testResource {
+	var (
+		sriovPolicy  *sriov.PolicyBuilder
+		sriovNetwork *sriov.NetworkBuilder
+		podBuilder   *pod.Builder
+	)
+
+	sriovPolicy = definePolicy(role, deviceType, nicVendor, pfName, vfRange)
+
+	sriovNetwork = defineNetwork(role, deviceType)
+
+	podBuilder = definePod(role, deviceType, workerNode)
+
+	return testResource{sriovPolicy, sriovNetwork, podBuilder}
+}
+
+func definePolicy(role, devType, nicVendor, pfName string, vfRange int) *sriov.PolicyBuilder {
+	var policy *sriov.PolicyBuilder
+
+	switch devType {
+	case "netdevice":
+		policy = sriov.NewPolicyBuilder(APIClient,
+			role+devType, NetConfig.SriovOperatorNamespace, role+devType, 2, []string{pfName}, NetConfig.WorkerLabelMap).
+			WithDevType("netdevice").
+			WithVFRange(vfRange, vfRange)
+	case "vfiopci":
+		if !(nicVendor == netparam.MlxDeviceID || nicVendor == netparam.MlxBFDeviceID) {
+			policy = sriov.NewPolicyBuilder(APIClient,
+				role+devType, NetConfig.SriovOperatorNamespace, role+devType, 2, []string{pfName}, NetConfig.WorkerLabelMap).
+				WithDevType("vfio-pci").
+				WithVFRange(vfRange, vfRange).
+				WithRDMA(false)
+		} else {
+			policy = sriov.NewPolicyBuilder(APIClient,
+				role+devType, NetConfig.SriovOperatorNamespace, role+devType, 2, []string{pfName}, NetConfig.WorkerLabelMap).
+				WithDevType("netdevice").
+				WithVFRange(vfRange, vfRange).
+				WithRDMA(true)
+		}
+	}
+
+	return policy
+}
+
+func defineNetwork(role, devType string) *sriov.NetworkBuilder {
+	network := sriov.NewNetworkBuilder(
+		APIClient, role+devType, NetConfig.SriovOperatorNamespace, tsparams.TestNamespaceName, role+devType).
+		WithMacAddressSupport().
+		WithIPAddressSupport().
+		WithStaticIpam()
+
+	return network
+}
+
+func definePod(role, devType, worker string) *pod.Builder {
```

Comment:
Can we use CreatePodsAndRunTraffic() instead?

Reply:
Not in this case. I have one more step between creating pod and running traffic, to modify arp table.